network.tf

```diff
@@ -8,13 +8,13 @@ resource "azurerm_virtual_hub_connection" "hub_connection" {
   remote_virtual_network_id = each.value.remote_virtual_network_id
   internet_security_enabled = try(each.value.internet_security_enabled, null)
   dynamic "routing" {
-    for_each = each.value.routing != null && length(each.value.routing) > 0 ? each.value.routing : []
+    for_each = each.value.routing != null && length(coalesce([each.value.routing], [])) > 0 ? [each.value.routing] : []
```

Comment:
It could be neater to shift this logic left into the variables and /or locals. Do the checks in the variables and / or locals and then the resources just needs to iterate and nothing else. The locals will provide either an empty collection or a collection with item(s) in it.

It could make the code easier to read.

Same goes for the for_each on this resource, the checks and balances could be shifted left. In this cased you already have a default of `{}` on that variable, so the checks should be unnecessary, but as Matt said, you could add `nullable = false` as a belt and braces approach.

For example, you could make changes similar to the ones in this branch: https://github.com/Azure/terraform-azurerm-avm-ptn-virtualwan/compare/azurerm_virtual_hub_connection_errs...example-azurerm_virtual_hub_connection_errs

Reply:
Thanks @jaredfholgate. The example was extremely helpful. Done.